source/OrderedTypes/PredicativeLFP.lagda

```diff
@@ -0,0 +1,1704 @@
+                  → b ∈ 𝓘nd
+  𝓘nd-is-c-closed = c-closed
+
+  𝓘nd-is-ϕ-closed : (a : ⟨ L ⟩)
+                  → (b : B)
+                  → (b , a) ∈ ϕ
+                  → ((b' : B) → (b' ≤ᴮ a → b' ∈ 𝓘nd))
+                  → b ∈ 𝓘nd
+  𝓘nd-is-ϕ-closed = ϕ-closed
+
+  𝓘nd-induction : (P : (b : B) → 𝓟 {𝓣} (b ∈ 𝓘nd))
+                → ((U : 𝓟 {𝓥} B) → (f : U ⊆ 𝓘nd)
+                 → ((x : B) → (u : x ∈ U) → f x u ∈ P x)
+                 → (b : B) → (g : (b ≤ᴮ (⋁ 【 β , U 】)))
+                 → c-closed U f b g ∈ P b)
+                → ((a : ⟨ L ⟩)
+                 → (b : B)
+                 → (p : (b , a) ∈ ϕ)
+                 → (f : (x : B) → (x ≤ᴮ a → x ∈ 𝓘nd))
+                 → ((x : B) → (o : x ≤ᴮ a) → f x o ∈ P x)
+                 → ϕ-closed a b p f ∈ P b)
+                → (b : B) → (i : b ∈ 𝓘nd) → i ∈ P b
+  𝓘nd-induction = Ind-induction
+
+  𝓘nd-recursion : (P : 𝓟 {𝓣} B)
+                → ((U : 𝓟 {𝓥} B)
+                 → (U ⊆ 𝓘nd)
+                 → (U ⊆ P)
+                 → (b : B) → (b ≤ᴮ (⋁ 【 β , U 】))
+                 → b ∈ P)
+                → ((a : ⟨ L ⟩)
+                 → (b : B)
+                 → (b , a) ∈ ϕ
+                 → ((x : B) → (x ≤ᴮ a → x ∈ 𝓘nd))
+                 → ((x : B) → (x ≤ᴮ a → x ∈ P))
+                 → b ∈ P)
+                → 𝓘nd ⊆ P
+  𝓘nd-recursion P = 𝓘nd-induction λ b → (λ _ → P b)
+
+  𝓘nd-is-initial : (P : 𝓟 {𝓣} B)
+                 → ((U : 𝓟 {𝓥} B)
+                  → (U ⊆ P)
+                  → ((b : B) → (b ≤ᴮ (⋁ 【 β , U 】))
+                  → b ∈ P))
+                 → ((a : ⟨ L ⟩)
+                  → (b : B)
+                  → (b , a) ∈ ϕ
+                  → ((b' : B) → (b' ≤ᴮ a → b' ∈ P)) → b ∈ P)
+                 → 𝓘nd ⊆ P
+  𝓘nd-is-initial {𝓣} P IH₁ IH₂ b b-in-𝓘nd = 𝓘nd-recursion P R S b b-in-𝓘nd
+   where
+    R : (U : 𝓟 {𝓥} B)
+      → U ⊆ 𝓘nd
+      → U ⊆ P
+      → (x : B) → x ≤ᴮ (⋁ 【 β , U 】)
+      →  x ∈ P
+    R U C₁ C₂ x o = IH₁ U C₂ x o
+    S : (a : ⟨ L ⟩)
+      → (x : B)
+      → (x , a) ∈ ϕ
+      → ((z : B) → z ≤ᴮ a → z ∈ 𝓘nd)
+      → ((z : B) → z ≤ᴮ a → z ∈ P)
+      → x ∈ P
+    S a x p f g = IH₂ a x p g
+
+\end{code}
+
+We now consider a restricted class of inductive definitions which we will call
+local. A local inductive definition ϕ is one such that the type
+
+  Σ b ꞉ B , (Ǝ a' ꞉ ⟨ L ⟩ , (b , a') ∈ ϕ × (a' ≤ a)
+
+is small.
+
+We then define an operator parameterized by local inductive definitions
+and prove that it is monotone. Finally, we show that any monotone endo map on
+a sup lattice determines a monotone operator and corresponding local
+inductive definition.
+
+\begin{code}
+
+module local-inductive-definitions
+        {𝓤 𝓦 𝓥 : Universe}
+        {B : 𝓥  ̇}
+        (L : Sup-Lattice 𝓤 𝓦 𝓥)
+        (β : B → ⟨ L ⟩)
+        (h : is-small-basis L β)
+       where
+
+ private
+  _≤_ = order-of L
+  ⋁_ = join-of L
+
+ open Joins _≤_ 
+ open is-small-basis h
+
+ S : (ϕ : 𝓟 {𝓤 ⊔ 𝓥} (B × ⟨ L ⟩)) → (a : ⟨ L ⟩) → 𝓤 ⊔ 𝓦 ⊔ 𝓥  ̇
```

Comment:
Can you remind me of the role that `S` plays? Hopefully that helps in coming up with a good name for it.

Reply:
Let's see... S is an intermediary in defining the monotone operator gamma. It's a generalized downsubset in some sense: it collects all the b's that are related (via phi) to some a' that is below a.

Reply:
Is `S` ever used/interesting on its own? If not, and it's essentially just part of a proof, then you could make this clear, put it all in a private block, and then the name `S` is not so bad anymore as it's clearly something temporary.

Reply:
It is used in another module.

Reply:
Maybe rename `S` to `_related-downset` then.